From 697ec6adf6193bd0e320d542e10d931997bc5d7b Mon Sep 17 00:00:00 2001
From: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
Date: Wed, 8 Nov 2023 08:53:49 +0530
Subject: [PATCH 3/3] Disable kernel page caching in nbdkit and mount.cifs

Issue:
In some platforms, when copying large media 9GB+ inside host,
kernel panic will happen inside BMC due to page alloc failure.

Fix:
Configured cache=none in nbdkit and mount.cifs

Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 src/smb.hpp                    |  2 +-
 src/state/activating_state.cpp | 13 ++++++++-----
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/src/smb.hpp b/src/smb.hpp
index 9e4a2fc..b694d48 100644
--- a/src/smb.hpp
+++ b/src/smb.hpp
@@ -23,7 +23,7 @@ class SmbShare
     {
         LogMsg(Logger::Debug, "Trying to mount remote : ", remote);
 
-        const std::string params = "sec=ntlmsspi,seal";
+        const std::string params = "sec=ntlmsspi,seal,cache=none";
         const std::string perm = rw ? "rw" : "ro";
         std::string options = params + "," + perm;
         std::string credentialsOpt;
diff --git a/src/state/activating_state.cpp b/src/state/activating_state.cpp
index 37da28f..aeaeb70 100644
--- a/src/state/activating_state.cpp
+++ b/src/state/activating_state.cpp
@@ -288,11 +288,14 @@ std::unique_ptr<resource::Process>
     ActivatingState::spawnNbdKit(interfaces::MountPointStateMachine& machine,
                                  const fs::path& file)
 {
-    return spawnNbdKit(machine, {},
-                       {// Use file plugin ...
-                        "file",
-                        // ... to mount file at this location
-                        "file=" + file.string()});
+    std::vector<std::string> params = {// Use file plugin ...
+				       "file",
+				       // ... to mount file at this location
+				       "file=" + file.string(),
+				       // Disable kernel page caching
+				       "cache=none"};
+
+    return spawnNbdKit(machine, {}, params);
 }
 
 std::unique_ptr<resource::Process>
-- 
2.34.1

